src/ibc_api/metadata.py

```diff
@@ -0,0 +1,87 @@
+"""Functions to fetch metadata about the available IBC datasets."""
+
+import json
+import os
+
+REMOTE_ROOT = (
+    "https://api.github.com/repos/individual-brain-charting/docs/contents"
+)
+
+LOCAL_ROOT = os.path.join(os.path.dirname(__file__), "data")
+os.makedirs(LOCAL_ROOT, exist_ok=True)
+
+SUBJECTS = [
+    f"sub-{sub:02d}" for sub in [1, 2, 4, 5, 6, 7, 8, 9, 11, 12, 13, 14, 15]
+]
+
+
+def _load_json(data_file):
+    """Read a given json file"""
+    with open(data_file, "r") as f:
+        data = json.load(f)
+
+    return data
+
+
+def select_dataset(data_type, metadata=None):
+    """Select metadata of the requested dataset"""
+    if metadata is None:
+        metadata = fetch_metadata()
+    try:
+        dataset = metadata[data_type]
+    except KeyError:
+        raise KeyError(
+            f"Dataset type {data_type} not found in IBC collection."
+        )
+    latest_version = find_latest_version(dataset)
+    dataset = dataset[latest_version]
+    return dataset
+
+
+def find_latest_version(dataset):
+    """Find the latest version of the dataset"""
+
+    latest_version_index = 0
+    latest_version = 0
+    for i, data in enumerate(dataset):
+        if data["version"] > latest_version:
+            latest_version = data["version"]
+            latest_version_index = i
+
+    return latest_version_index
+
+
+def fetch_remote_file(
+    file="datasets.json",
+    remote_root=REMOTE_ROOT,
+    local_root=LOCAL_ROOT,
+):
+    """Fetch a file from the IBC docs repo"""
```

Comment:
same here (alternatively, you can add a leading underscore to the function name)